src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -193,62 +202,78 @@ class boss_eadric : public CreatureScript
                 damage = 0;
                 EnterEvadeMode();
                 me->setFaction(35);
-                bDone = true;
+                if (!_defeated)
```

Comment:
better put those
````c++
EnterEvadeMode();
me->setFaction(35);
````
 inside the if, otherwise they would be called several times.